MaxText/layers/models.py

```diff
@@ -16,731 +16,26 @@
 # pylint: disable=arguments-differ
 # pylint: disable=no-name-in-module
 
-from typing import Any, Optional
-import functools
+from typing import Optional
 
-import jax
 import jax.numpy as jnp
-from jax.ad_checkpoint import checkpoint_name
 from jax.sharding import Mesh
 
 from flax import linen as nn
-from flax.linen.partitioning import ScanIn
 
-from MaxText.common_types import DecoderBlockType, Config, MODEL_MODE_TRAIN, MODEL_MODE_PREFILL, MODEL_MODE_AUTOREGRESSIVE, DECODING_ACTIVE_SEQUENCE_INDICATOR
-from MaxText import max_logging
+from MaxText.common_types import DecoderBlockType, Config, MODEL_MODE_TRAIN, MODEL_MODE_AUTOREGRESSIVE, DECODING_ACTIVE_SEQUENCE_INDICATOR
 from MaxText.inference import page_manager
-from MaxText.layers import linears
-from MaxText.layers import quantizations
-from MaxText.layers import pipeline
-from MaxText import maxtext_utils
 from MaxText import multimodal_utils
-from MaxText.layers.attentions import Attention
-from MaxText.layers.normalizations import rms_norm
-from MaxText.layers.embeddings import positional_embedding_as_linen, Embed
+from MaxText.layers.decoders import Decoder
```

Comment:
Shall we just import module instead of class? Also mentioned in python guide, [FYI](https://screenshot.googleplex.com/9DMo3p58g4e4g67). Also, could you double check other imports?

However, I see this import is widely used `from MaxText.common_types import ....` in the repo.